cmd/nvidia_gpu/nvidia_gpu.go

```diff
@@ -208,48 +207,40 @@ func (ngm *nvidiaGPUManager) Serve(pMountPath, kEndpoint, pEndpointPrefix string
 	for {
 		pluginEndpoint := fmt.Sprintf("%s-%d.sock", pEndpointPrefix, time.Now().Unix())
 		pluginEndpointPath := path.Join(pMountPath, pluginEndpoint)
+		glog.Infof("starting device-plugin server at: %s\n", pluginEndpointPath)
+		lis, err := net.Listen("unix", pluginEndpointPath)
+		if err != nil {
+			glog.Fatalf("starting device-plugin server failed: %v", err)
+		}
+		ngm.grpcServer = grpc.NewServer()
+		pluginapi.RegisterDevicePluginServer(ngm.grpcServer, ngm)
+
 		var wg sync.WaitGroup
 		wg.Add(1)
 		// Starts device plugin service.
 		go func() {
 			defer wg.Done()
-			glog.Infof("starting device-plugin server at: %s\n", pluginEndpointPath)
-			lis, err := net.Listen("unix", pluginEndpointPath)
-			if err != nil {
-				glog.Fatalf("starting device-plugin server failed: %v", err)
-			}
-			grpcServer := grpc.NewServer()
-			pluginapi.RegisterDevicePluginServer(grpcServer, ngm)
-			ngm.Lock()
-			ngm.grpcServer = grpcServer
-			ngm.Unlock()
-			ngm.grpcServer.Serve(lis)
+			glog.Errorf("device-plugin server stopped serving: %v", ngm.grpcServer.Serve(lis))
```

Comment:
nit: suggest we change this to:
err := ngm.grpcServer.Serve(lis)
glog.Errorf("device-plugin server stopped serving: %v", err)
This makes the functionality of this go routine more obvious.

Reply:
Done.